src/Migrations/M032MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,360 @@
+    private const SOURCE_TWITTER = 'twitter';
+    private const SOURCE_X = 'x';
+
+    private const SOURCE_PATTERNS = [
+        self::SOURCE_INSTAGRAM => '/^(https?\:\/\/)?(www\.)?instagram\.com\/.+$/i',
+        self::SOURCE_FACEBOOK => '/^(https?\:\/\/)?(www\.)?facebook\.com\/.+$/i',
+        self::SOURCE_TWITTER => '/^(https?\:\/\/)?(www\.)?(twitter\.com)\/.+$/i',
+        self::SOURCE_X => '/^(https?\:\/\/)?(www\.)?x\.com\/.+$/i',
+    ];
+
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
+    {
+        try {
+            // Get the list of posts using Media blocks.
+            $posts = Utils\Functions::get_posts_using_specific_block(
+                Utils\Constants::BLOCK_SOCIAL_MEDIA,
+                Utils\Constants::ALL_POST_TYPES
+            );
+
+            // If there are no posts, abort.
+            if (!$posts) {
+                return;
+            }
+
+            echo "Social Media block migration in progress...\n"; // phpcs:ignore
+
+            $parser = new WP_Block_Parser();
+
+            // Variable to store the current post ID
+            $current_post_id = null;
+
+            // Get all the blocks of each post.
+            foreach ($posts as $post) {
+                if (empty($post->post_content)) {
+                    continue;
+                }
+
+                $current_post_id = $post->ID; // Store the current post ID
+
+                echo 'Parsing post ', $post->ID, "\n"; // phpcs:ignore
+
+                $blocks = $parser->parse($post->post_content);
+
+                if (!is_array($blocks)) {
+                    throw new \Exception("Invalid block structure for post #" . $post->ID);
+                }
+
+                foreach ($blocks as &$block) {
+                    // Check if the block is valid.
+                    if (!is_array($block)) {
+                        continue;
+                    }
+
+                    // Check if the block has a 'blockName' key.
+                    if (!isset($block['blockName'])) {
+                        continue;
+                    }
+
+                    // Check if the block is a Social Media block.
+                    if ($block['blockName'] !== Utils\Constants::BLOCK_SOCIAL_MEDIA) {
+                        continue;
+                    }
+
+                    // Check if the embed is a Facebook page, in which case we do nothing.
+                    if (isset($block['attrs']['embed_type']) && $block['attrs']['embed_type'] === 'facebook_page') {
+                        continue;
+                    }
+
+                    // If not, we get the social media URL.
+                    $social_media_url = self::get_social_media_url($block);
```

Comment:
I think we don't need the `get_social_media_url` function:

 ```suggestion
                    $social_media_url = $block['attrs']['social_media_url'] ?? null
```